### PR TITLE
Old Vendotron Events

### PR DESCRIPTION
**Pull request by GreatBigFailure**
This PR adds two random events, both act as methods of spawning the "old vendotron" on the station.
**Sprites are by Megathaum who is very talented and I heard has big muscles**
**Huuuuuge muscles**

capitalismcrash

### Old Vendotron Events
**Rod Version**
The vendor acts like an immovable rod but loses momentum after a few walls and stops, letting you use the vending machine. The event has 1/3rd of the chance to trigger as meteors do.

**Teleport Version**
The vendor is spawned through a bluespace tear at a random location on the station. This is completely non-destructive. The event has the same chance to trigger as meteors.

### Old Vendotron
The vending machine is stocked randomly from a large list of different items. Each item has a base price which is then randomly increased or decreased for variety.

**The stock**
I tried to make sure all the items possible were useful to more than just traitors, and also intentionally avoided anything that was lethal in a straight forward way, so no eguns. Even the items that are usually only available to antags were picked because they could be used in interesting non-antag ways. 
The potential stock is separated into common, uncommon, and rare stock.

Common stock: A mixture of mundane to hard-but-reliable to get items. There are some exceptions but nothing extreme. Examples are: a corgi puppy, maint pills, a hookshot.

Uncommon stock: Things you would find in a vault, minor antag restricted items, some xenoarch small finds, things you'd steal from the vault safe. Examples are: a gremlin, a compressed matter implant, a bottle of creatine pills.

Rare stock: Things you would need to pray for or are otherwise bussed in, the "big" find at the end of a vault, the one item you look for at spessmart. Examples are: A flawless bluespace crystal, a gingerbread egg, and a multi-person wheelchair.

**How Stock Works**
tl;dr: 6-18 items in stock rolled: 70% common -> 80% uncommon -> rare. Then additional uncommon rolls based on round pop.

Between 6-18 stock are available, randomly. For each stock you have a 70% chance to roll something from the common list, otherwise you use the uncommon list, which further has a 20% chance to use the rare list instead.
The stock is increased by 1 + 10% of the current round population, this bonus stock skips right to the uncommon list.

**Pricing**
Each item has a base price which is further modified by anywhere between -30% to 30% when the stock is rolled. 
Example prices:
- Donkpocket box, common, 50
- Tournament lance, common, 250
- Amorphous mask, uncommon, 300
- Chameleon jumpsuit, uncommon, 250
- Portal gun, rare, 1100
- Sparebear pelt, rare, 800

**Wait what's all this stuff in here about platesplatesplates and-**
Don't look at that, be quiet

### More Stock, Ideaguy me
The PR is now ready for review but I would like to keep adding stock during that time. If something pops into your head please suggest it.

:cl:
 * rscadd: The old vendotron is coming to get your credits, either via a teleportation event or a rod-like crash event.

### CODE REVIEW

**Comment by Kurfursten** [comment]
> Oh boy it's the vending machine event again
> My department was destroyed just like with a rod
> But hey, now the miners can by new guns and antag items for 600 credits a pop, can't wait to get killed by emitter glasses

Maybe it won't be horrible but I don't think we have enough of something resembling a serious economy to have an event based around money.

The anti-tamper features are cute though.

File: code/datums/helper_datums/command_alerts.dm [hunk 1]
```diff
@@ -740,3 +740,7 @@ The access requirements on the Asteroid Shuttles' consoles have now been revoked
 	name = "Ancient Cryogenic Pod"
 	alert_title = "Abnormal Life Sign Report"
 	message = "An abnormal life sign has been detected in promiximity of the station. Long range scans determine signal to be human life. Approach with caution."
+
+/datum/command_alert/old_vendotron
+	alert_title = "Odd Machine Debris"
+	message = "A large chunk of debris is on a collision course with your station. Moderate damage is expected at its current velocity. The nature of the debris is unknown, however our sensors are picking up what sounds like the feint jingling of coins."
```
Comment (boy2mantwicethefam) [inline code/datums/helper_datums/command_alerts.dm]: faint

**Comment by GreatBigFailure** [comment]
@Kurfursten I'm still tinkering with the specifics on how destructive it can be. I'm likely going to more than halve the tiles it can travel through, since I never intended it to be able to get as far as medbay like in the screenshot. 
Since only one event will fire at once and it has the same chance to fire as meteors, it's a net loss in station explosion. 
As for the economy issue, I kind of agree. Hell the vault cash alone completely changed how I had to price things. Still, I've been wanting to add more uses to credits for ages. Credits having purpose vs credits being even remotely balanced is a chicken and egg scenario.
I have no idea what anti-tamper features you may be talking about

**Comment by Kurfursten** [comment]
>net loss of destruction

Not at all, think of it this way: consider all these events have equal chances (this is purely for the sake of example

>before 
(destructive: 1/7) meteors (not destructive: 6/7) comms blackout, greytide virus, spam, vermin, wallrot, appendix burst

>after
(destructive: 2/8) meteors, rod-vendor (not destructive: 6/8) comms blackout, greytide virus, spam, vermin, wallrot, appendix burst

You aren't replacing a meteor event, you are increasing the number of destructive possibilities. This means there's another breaching event that might take the place of, say, appendicitus. I'm not saying that this is a dealbreaker or there shouldn't be breaching events, but don't act like it isn't one or that it doesn't increase the odds.

**Comment by GreatBigFailure** [comment]
I hadn't looked at it that way.
Regardless, I'm tinkering with the amount of destruction it's capable of. In this draft it chooses between 10 and 25 dense tiles it can pass through, now it's at 5 to 10. I want it to consistently get past solars and what not but not be even close to a shuttle call, just a broken wall or two.
Really though if people don't like the breaching I'm not married to it. This is intended to be a positive event, which we have very few of.

**Comment by gurfan** [comment]
I dislike the current stock. After this thing lands the first thing people are going to do is crack open the vault and buy the most annoying, gunky items (flawless crystal, amorphous masks, portal gun, the list goes on). This feels like a lot like spessmart, yet unlike spessmart the vendor comes to the station and won't murder half the crew. I think it would be a lot more better if the items in here were more similar to the trader items in that they can assist people with their job. (not that they should be selling the same items traders do) There are a lot of unused tools in the code (i.e. ranged atmos analyzer) that I think could be a better fit. Tier 4 parts might also be more useful. 
tldr the stock is filled with too many dumb toys that won't make the round any better

**Comment by KillFurries** [comment]
I'd look into stealing code from the trade vendor. Crashing a vending machine into the station can cause some problems, you could instead have the trade window pop up on the station with a Resident Evil 4 style trader offering goods. You still get items that could make the round more interesting but it's less disruptive. All you have to do is teleport the vendor in and then teleport it out.
Or have both. Whatever makes rounds more interesting.

**Comment by GreatBigFailure** [comment]
@gurfan While testing I've been spawning a lot of them and the stock actually comes off as much more mundane than I expected, to the point that I'm likely to boost the uncommon chance to 25 or 30 from 20. I get the concern though, which is the specific reason I made this a draft.
Cracking open the vault isn't something I can fix with this though. I however don't think we should be basing all future content off the idea that someone years ago decided the vault should have nearly 7000 credits. If the vault causes problems for all other content related to credits then it should be changed, but we don't have enough credit based content (because the vault exists, it seems) to warrant it.

Either way, please suggest more items you'd like to see in it. Upgrade components are a really good idea, I'll add those now. 
Keep in mind I intentionally made every item in each rarity list have the same chances as its peers so that more additions would directly dilute the chances of getting the jackpots.

@KillFurries I don't know how I feel about it just wizard teleporting onto the station but people seem concerned with it imitating a rod for a few tiles so if zero destruction is preferred I'll go with that, or something similar.
For a variety of reasons I'm keeping it as a vendor rather than the new trader NPC. Partially due to sunk cost fallacy but also because [redacted]

**Comment by GreatBigFailure** [comment]
Here's some examples of the results of the event, currently.
Two points: This is with possible walls destroyed at 3 to 8, I since chanced it to 4 to 6. I don't want instances like example 4 to be at all common.
Also ignore the prices as the RNG adjuster isn't working correctly.

1. Came in from the north, damaged perma, stopped before the brig proper.
crash1
 
2. Came from the south, broke right through the CE locker but didn't do much damage
crash2

3. South east, ffffuck virology
crash3

4. South west, tore through cargo, the hall, and the bridge. Vented the main hallway. Were the donk pockets worth it?
crash4

5. North east, nothing of value was lost
crash5

**Comment by gurfan** [comment]
> Cracking open the vault isn't something I can fix with this though. I however don't think we should be basing all future content off the idea that someone years ago decided the vault should have nearly 7000 credits. If the vault causes problems for all other content related to credits then it should be changed, but we don't have enough credit based content (because the vault exists, it seems) to warrant it.

The prices of the items or the rarity of them isn't the issue here. A lot of the items that can be sold should not be easily accessible by the crew (money is not difficult to obtain even without the vault). 

Some examples of useful and less disrupting items that I think might be good additions:
- Tier 4 Parts
- A special limitless RPED
- Ranged Atmos Analyzer
- Certain Slime Extracts (to upgrade things)
- Rare Mecha Boards/Parts
- An advanced lockbox 

~~Overall I don't feel that the vendor with its current selection of items would improve the round in any way.~~
I think that a lot of the items would really gunk up a round.

**Comment by DamianX** [comment]
the obvious solution is to remove the vault money

**Comment by GreatBigFailure** [comment]
@gurfan I disagree on the last point, though admittedly a lot of the items are dummy prizes meant to add variety between events (One might have nothing but aesthetic common items, the next might be spessmart 2.0). Keep in mind the whole "easily accessible" thing is an over statement. After spawning dozens of them I haven't seen any of the items that I considered risky to add. The RNG of the thing is so layered for a reason. 
I'll add the items you suggested, I meant to add the slime extracts before but forgot. Thanks for the list.

@DamianX I'd at least like to see it be more complicated than "hack door, receive infinite credits". It's very easy to have the vault money within the first minute or two of a round starting, which is silly. I think it should exist in some capacity but the way it currently does just ruins anything related to credits. Traders just need to pretend it doesn't exist since it overwrites their whole shtick. This is coming from someone who very frequently does go steal the vault cash within a minute or two.

**Comment by GreatBigFailure** [comment]
Okay I think it's technically done. I would like more stock first though. Please make more suggestions.
### **Updates**
- In addition to the previous crash version there is a second event that just teleports the vending machine onto the station with a bit of fanfare. The crash event had its likelihood of triggering cut down to 1/3rd of what it was, the teleport version has the same chance the crash used to (same chance as a meteor event). Both events have separate admin bus buttons to trigger them.
- A bunch of new stock was added based on suggestions, some broken stock was fixed. I still need more though.
- Uncommon stock now has a 30% (up from 20%) chance to be rolled. The vendors were too mundane before.
- Common stock now has a chance to have multiples of itself stocked (50% chance for 2, 25% chance for 3). Uncommon stock has the same mechanic with lower chance (15% chance for 2).
- Everything works now, probably
- That's it, don't look at anything else. "Ghetto nightmare" is just a rumor, it doesn't exist. Be quiet.

**Comment by Eneocho** [comment]
One guy on the thread suggested to use the new UI that the vox vendor got.

**Comment by DriftingSkies** [comment]
The trader metaclub is once again under ATTACK!

Really though, good idea, but all I ask is don't have trader goods in the vendotron. It makes people want to exchange goods even less!

**Comment by Eneocho** [comment]
I'd keep the traderstuff on the vendor anyway. The odds of the vendor items appearing are low enough, and you have to get lucky and get the vendortron first anyway. Traders are more reliable to get the stuff, even if it means dealing with (((vox))).

**Comment by GreatBigFailure** [comment]
The amount of trader items in there is very limited and they all cost noticably more than if you had got them from the trader.
Like Eneocho said, the event needs to spawn, the items need to show up, the price needs to roll well, you need to beat the greytide and sec to the vendor (I foresee sec confiscating vendors fairly often) and you need to find/have access to the area it spawned in to begin with.

**Comment by ghost** [comment]
whats this about crowbaring the vending machine pulling a soviet russia on you with it?

**Comment by GreatBigFailure** [comment]
What? No that's just a rumor, a lie. You're looking at a different pr. You're confused.

**Comment by BomberBro** [comment]
I've been asked to give my opinion as a pessimistic husk, so here it is:
>/obj/item/clothing/mask/stone = 12000,

No. This should be obvious.
>/obj/item/clothing/head/bearpelt/real/spare = 800,

My main complaint for the pelt is that the sprite is awful, last time I checked there was even a huge section of white pixels on one of the directions that nobody removed when making it. Also adding the option for a potential spare at a cost any assistant can get fairly easily might be a "bad call".

As for the rest, depending on how far the vendotron goes into the station it effectively just doubles the weight of rods, except this time with a treat at the end. I see in one of the screenshots it hit the bridge, and in another it's at sec EVA, but it traveled from the **opposite side of the station**. Rods aren't terribly interesting and if this has the capability to cut through half the station it'll carry those some drawbacks. Also, I dislike the emag-only interactions, but since that's what vending machines have by default I guess it's thematic.

**Comment by GreatBigFailure** [comment]
@BomberBro Thank you husk man

>Stone mask

Surprised that's the first time someone mentioned that. With that said it's extremely hard to get. Even if you rolled it with a maximum discount of -30% it still costs 8400, more than the whole vault plus what most people make in wages all shift. I can make it even pricier but I think we'll get at least one round based around some kind of vampire fundraiser or something. I'm not married to the idea though and even dislike the fact that 99% of the time the mask is rolled it's basically just a dud rare item that could have been more fun since it's not going to be bought. Especially since sec will likely confiscate the vendor if the mask shows up.

>Sprite is awful

Is it? I've barely looked at it, thanks for the heads up. 

>Doubles the weight of rods

Which is why I split it into two events and made the rod version much rarer. The second event just harmlessly teleports it onto the station with no negatives involved. For actual numbers:
Rad storm: 50
Meteors: 15
Rod: 15
Vendor rod: 5
Vendor teleport: 15

Additionally I significantly reduced how far the rod version will travel. It now, in most cases, barely breaches the station. The original numbers were because I misunderstood what the rod considered dense.

>Emag only

There are multiple layers of "That doesn't exist" here. One layer is that any interactions an emag may or may not have with it are not exclusive and can happen through other means, the weights would, hypothetically, just be different for an emag vs those other means. The second layer is that what you said as well as the first layer are just rumors, do not exist, never have or will, and I need you to stop feeding the rumor mill because the young and impressionable generation might believe in such lies as "ghetto nightmare" and even "neo-ultra-capitalism mode", neither of which exist.

**Comment by BomberBro** [comment]
The stone mask would probably just be free antag for miners, which is lame. It's still easy to get a lot of money as a miner so I don't think it fits, plus if sec is (justifiably) taking the vendor because it's contraband then that kind of nullifies the event.

**Comment by GreatBigFailure** [comment]
I'd like to know how plausible 12,000  credits is for a miner but fair enough on the other points. I'll remove it.

**Comment by GreatBigFailure** [comment]
Replaced the stone mask with a significantly cheaper black gift.

File: code/modules/events/old_vendotron_event/old_vendotron.dm [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+	prices[chosenStock] = stockPrice
+
+/obj/machinery/vending/old_vendotron/proc/priceRandomizer(var/thePrice = 0)
+	thePrice = rand(thePrice * 0.7, thePrice * 1.3)	//30% cheaper or pricier, totally random for SPICE
+	return thePrice
+
+
+//The end///////////////
+
+
```
Comment (Kurfursten) [inline code/modules/events/old_vendotron_event/old_vendotron.dm]: delete whitespace, nothing prevents codediving anyway

**Comment by GreatBigFailure** [comment]
Oh god oh fuck did it work? I banish thee git from my house. BEGONE

**Comment by Eneocho** [comment]
>Replaced the stone mask with a significantly cheaper black gift.

Black gifts are currently garbage. A poisoned apple, a bomb, a cult 2 book and an emag. It needs more stuff.
